com.unity.shadergraph/Editor/GraphDeltaRegistry/FunctionDefinitions/StandardDefinitions/NormalReconstructZNode.cs

```diff
@@ -0,0 +1,34 @@
+using System.Collections.Generic;
+using static UnityEditor.ShaderGraph.Registry.Types.GraphType;
+
+namespace com.unity.shadergraph.defs
+{
+
+    internal class NormalReconstructZNode : IStandardNode
+    {
+        public static FunctionDescriptor FunctionDescriptor => new(
+            1,
+            "NormalReconstructZ",
+            @"
+{
+    reconstructZ = sqrt(1.0 - saturate(dot(In.xy, In.xy)));
+    normalVector = float3(In.x, In.y, reconstructZ);
+    Out = normalize(normalVector);
+}",
+            new ParameterDescriptor("In", TYPE.Vector, Usage.In),
```

Comment:
type should be Vec2

Reply:
In the old code, the input port can connect to vectors and int. If we set the TYPE to vec2, will the input port allow the rest of the vectors?

Reply:
Any other data type that I connect to it in the old system gets automatically cast to vec2.  And in the old code the port is explicitly set to be a vec2:  [Slot(0, Binding.None)] Vector2 In,